components/match2/wikis/splitgate/match_summary.lua

```diff
@@ -0,0 +1,234 @@
+				:body(CustomMatchSummary._createBody(match))
+
+	-- comment
+	if match.comment then
+		local comment = MatchSummary.Comment():content(match.comment)
+		comment.root:css('display', 'block'):css('text-align', 'center')
+		matchSummary:comment(comment)
+	end
+
+	-- footer
+	local vods = {}
+	for index, game in ipairs(match.games) do
+		if game.vod then
+			vods[index] = game.vod
+		end
+	end
+
+	match.links.lrthread = match.lrthread
+	match.links.vod = match.vod
+	if not Table.isEmpty(vods) or not Table.isEmpty(match.links) then
+		local footer = MatchSummary.Footer()
+
+		-- Game Vods
+		for index, vod in pairs(vods) do
+			footer:addElement(Template.safeExpand(frame, 'vodlink', {
+				gamenum = index,
+				vod = vod,
+				source = vod.url
+			}))
+		end
+
+		-- Match Vod + other links
+		local buildLink = function (linkType, link)
+			local linkData = _LINK_DATA[linkType]
+			if not linkData then
+				mw.log('linkType "' .. linkType .. '" is not supported by Module:MatchSummary')
+			else
+				return '[['..linkData.icon..'|link='..link..'|15px|'..linkData.text..']]'
+			end
+		end
+
+		for linkType, link in pairs(match.links) do
+			footer:addElement(buildLink(linkType,link))
+		end
+
+		matchSummary:footer(footer)
+	end
+
+	return matchSummary:create()
+end
+
+function CustomMatchSummary._addFlexRow(wrapper, contentElements, class, style)
+	local node = htmlCreate('div'):addClass('brkts-popup-body-element')
+	if not Logic.isEmpty(class) then
+		node:addClass(class)
+	end
+	for key, val in pairs(style or {}) do
+		node:css(key, val)
+	end
+	for _, element in ipairs(contentElements) do
+		node:node(element)
+	end
+	return wrapper:node(node)
+end
+
+function CustomMatchSummary._createHeader(match)
+	local header = MatchSummary.Header()
+
+	header:leftOpponent(CustomMatchSummary._createOpponent(match.opponents[1], 'left'))
+	      :leftScore(CustomMatchSummary._createScore(match.opponents[1]))
+	      :rightScore(CustomMatchSummary._createScore(match.opponents[2]))
+	      :rightOpponent(CustomMatchSummary._createOpponent(match.opponents[2], 'right'))
+
+	return header
+end
+
+function CustomMatchSummary._createScore(opponent)
+	return OpponentDisplay.BlockScore{
+		isWinner = opponent.placement == 1 or opponent.advances,
+		scoreText = OpponentDisplay.InlineScore(opponent),
+	}
+end
+
+function CustomMatchSummary._createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or (match.date ~= _EPOCH_TIME_EXTENDED and match.date ~= _EPOCH_TIME) then
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not epoch=0, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	for _, game in ipairs(match.games) do
+		if game.map then
+			body:addRow(CustomMatchSummary._createMapRow(game))
+		end
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createOpponent(opponent, side)
```

Comment:
maybe move this above the `_createBody` function
(so that the subfunctions for the header are all together and before the body part)